lib/configuration.rb

```diff
@@ -109,4 +109,57 @@ def get_optional_class_instance(keyword)
       return Object.const_get(class_opts[:class]).new(**class_opts[:opts])
     end
   end
+
+  class FlexyEnvVariable
+    attr_reader :var_name
+
+    def initialize(var_name)
+      @var_name=var_name
+    end
+    def ==(other)
+      self.class===other &&
+        @var_name==other.var_name
+    end
+    def to_s
+      ENV[@var_name]
+    end
+    
+    alias :eql? :==
+    alias :to_str :to_s
+  end
+
+  class FlexyEnvFile
+    @@files = Hash.new
+    attr_reader :var_name
+
+    def initialize(var_name)
+      @var_name=var_name
+    end
+    def ==(other)
+      self.class===other && 
+        @var_name==other.var_name
+    end
+    def to_s
+      file=@@files[@var_name]
+
+      # file has been deleted on filesystem
+      if !file.nil? && !File.file?(file.path)
```

Comment:
just a nitpick, I think it is better readable to use `file` instead of `!file.nil?`

Things look good to me so I hope we can merge soon.

Reply:
:+1: 
Just a new information about Ruby received...:)

Reply:
In ruby only `nil` and `false` count as a falsy value and everything else as truish... including empty strings/arrays, `0`, etc.
Just for completeness.